### PR TITLE
Add shadowsocks 2022 / quic sniffer

### PR DESCRIPTION
**Pull request by nekohasekai**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,19 +9,23 @@ require (
...
 	github.com/gorilla/websocket v1.5.0
 	github.com/lucas-clemente/quic-go v0.27.0
+	github.com/marten-seemann/qtls-go1-16 v0.1.5
+	github.com/marten-seemann/qtls-go1-17 v0.1.1
+	github.com/marten-seemann/qtls-go1-18 v0.1.1
 	github.com/miekg/dns v1.1.49
 	github.com/pelletier/go-toml v1.9.5
 	github.com/pires/go-proxyproto v0.6.2
 	github.com/refraction-networking/utls v1.1.0
+	github.com/sagernet/sing v0.0.0-20220513023102-da424ddd3fb2
```
Comment (yuhan6665) [inline go.mod:19]: 感谢世界大佬pr！
quic sniffer 没啥大问题 我从 fly 那边搬了几个 fix 你看一下 https://github.com/XTLS/Xray-core/commits/quic-sniffer 没啥问题我就合了
ss 2022 有没有可能重新组织一下 把协议相关的代码单独放一个库（比如 https://github.com/v2fly/ss-bloomring ）主要是目前这个做法会把一些不相关依赖引入 *ray core
另外能否提供一份配置示例。。再次感谢大佬们的工作！

**Comment by nekohasekai** [comment]
Server:

```json
{
  "log": {
    "loglevel": "debug"
  },
  "inbounds": [
    {
      "listen": "0.0.0.0",
      "port": 1234,
      "protocol": "shadowsocks_2022",
      "settings": {
        "method": "2022-blake3-aes-128-gcm",
        "key": "sKUtO08onS3imJ68LTtgBNHyf/hMvizYHrFX420j22A="
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom"
    }
  ]
}
```

Client:

```json
{
  "log": {
    "loglevel": "debug"
  },
  "inbounds": [
    {
      "listen": "127.0.0.1",
      "port": 1080,
      "protocol": "socks",
      "settings": {
        "auth": "noauth",
        "udp": true,
        "ip": "127.0.0.1"
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "shadowsocks_2022",
      "settings": {
        "address": "127.0.0.1",
        "port": 1234,
        "method": "2022-blake3-aes-128-gcm",
        "key": "sKUtO08onS3imJ68LTtgBNHyf/hMvizYHrFX420j22A="
      }
    }
  ]
}
```

**Comment by nekohasekai** [comment]
Protocol name changed to `shadowsocks-2022`.

**Comment by yuhan6665** [comment]
感谢世界大佬！
我之后再研究一下有什么可以改进的地方。。

**Comment by nekohasekai** [comment]
> @nekohasekai Hi, even two years ago, at the pull request was merged, your repo is under GPL-3.0 or later, and it's incompatible with xray-core.
> 
> The library should be removed and never be depend on as long as this repo is still under MPL.
> 
> See:
> 
> * [License issue #3272](https://github.com/XTLS/Xray-core/issues/3272)
> 
> Can you consider withdrawing your granted permission now?

I created it with absolutely no legal considerations in mind, and I'm no longer involved in maintaining the project, so anyone can raise a PR to remove the functionality.

**Comment by nekohasekai** [comment]
> @nekohasekai Can you explicitly say 'Yes', to my question about withdrawing the permission?
> 
> Would you like to withdraw?

Yes

**Pull request merged**